internal/strcase/context.go

```diff
@@ -34,12 +37,20 @@ func New(acronyms map[string]string) Context {
 			Replacement: replacement,
 		}
 	}
-	return Context{
+	return context.WithValue(ctx, ctxKey{}, &Context{
 		acronyms: parsedAcronyms,
-	}
+	})
 }
 
-func (ctx Context) rangeAcronym(full string, pos int) (*acronymRegex, string) {
+func FromContext(ctx context.Context) *Context {
+	s, _ := ctx.Value(ctxKey{}).(*Context)

```

Comment:
1. DO NOT mix the contex.Context with strcase object. The strace package need support general use cases without context.Context.
2. readability: return default empty context `return &Context{}` if not found.